doc/source/ray-air/key-concepts.rst

```diff
@@ -0,0 +1,129 @@
+AIR Key Concepts
+================
+
+Here, we cover the main concepts that AIR is consisted of.
+
+.. contents::
+    :local:
+
+
+Preprocessors
+-------------
+
+Preprocessors are primitives that can be used to transform input data into features.
+
+A preprocessor can be fitted during Training, and applied at runtime in both Training and Serving on data batches in the same way.
+
+Preprocessors operate on :ref:`Ray Datasets <datasets>`.
+
+.. code-block:: python
```

Comment:
can we please add these snippets into scripts in a `doc_code` folder and test that in CI? We don't want code directly in docs anymore (unless using notebooks). Let's start this off the right way! :D